x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -139,12 +139,17 @@ fromCommand
 
 indexPattern
     : (clusterString COLON)? indexString
+    | {this.isDevVersion()}? (clusterString COLON)? indexString (CAST_OP selectorString)?
     ;
 
 clusterString
     : UNQUOTED_SOURCE
     ;
 
+selectorString
+    : UNQUOTED_SOURCE
```

Comment:
Should we allow quoting selector?

Reply:
I tried to simply follow how the definition for the clusterString are laid out. I don't have too much of an opinion on quoting the selector or not. The selector contents can only ever be `data`, `failures`, or `*`. I don't know what quoting them separately from the rest of the index pattern might buy us.